test_vector_generator/src/awses_test_vectors/manifests/full_message/encrypt.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""
+AWS Encryption SDK Encrypt Message manifest handler.
+
+Described in AWS Crypto Tools Test Vector Framework feature #0003 AWS Encryption SDK Encrypt Message.
+"""
+import binascii
+import json
+import os
+import uuid
+
+import attr
+import aws_encryption_sdk
+import six
+from aws_encryption_sdk.identifiers import AlgorithmSuite
+from aws_encryption_sdk.key_providers.base import MasterKeyProvider
+
+from awses_test_vectors.internal.defaults import ENCODING
+from awses_test_vectors.internal.util import (
+    algorithm_suite_from_string_id,
+    dictionary_validator,
+    file_reader,
+    file_writer,
+    iterable_validator,
+    membership_validator,
+    validate_manifest_type,
+)
+from awses_test_vectors.manifests.full_message.decrypt import DecryptMessageManifest, DecryptTestScenario
+from awses_test_vectors.manifests.keys import KeysManifest
+from awses_test_vectors.manifests.master_key import MasterKeySpec, master_key_provider_from_master_key_specs
+
+try:  # Python 3.5.0 and 3.5.1 have incompatible typing modules
```

Comment:
This appears in a lot of source files. Would it make sense to have an awses_test_vectors.typing import?

Reply:
Not really, at least not without violating import best practices (we could technically put everything in `mypy_types` and do `from mypy_types import *`, but that's rightfully considered an antipattern). We need these imports when the `mypy` static analyzer runs, but CPython 3.5.0 and 3.5.1, both of which we support, have some inconsistencies in the `typing` module. That's ok, because the `mypy` analyzer can't run in those versions anyway, but we need the code itself to run under those versions.